Add support for creating SIB resources and WS Policy binding attributes

gahoekstraibm opened the pull request:
Hi, 

We are considering to use WDR to enable continuous deployment in our project. However, we noticed some limitations, where I did some enhancements. You may consider to make these enhancements part of the main WDR library:

Add support for creating SIB queues, J2C connection factories and J2C activation specs

Add support for WS Policy binding attributes, to e.g. specify a username/password for Basic Auth HTTP policy

gahoekstraibm committed:
Create bla.py

Created initial python object to handle BLAs and Assets

gahoekstraibm committed:
Update manifest.py

Added support for creating SIB Queue destinations

gahoekstraibm committed:
implemented new checksum generation for EARs

gahoekstraibm committed:
Update util.py

gahoekstraibm committed:
Update util.py

gahoekstraibm committed:
Update util.py

gahoekstraibm committed:
Update util.py

gahoekstraibm committed:
Update util.py

gahoekstraibm committed:
Update util.py

gahoekstraibm committed:
Update util.py

gahoekstraibm committed:
Update util.py

gahoekstraibm committed:
Update manifest.py

Reverted the use of the new EAR checksum implementation, as it might not solve our problem anyway

gahoekstraibm committed:
Added constructor for J2CAdminObject

gahoekstraibm committed:
Update manifest.py

Added hook to construct a J2CConnectionFactory

gahoekstraibm committed:
Extended support for WS policy bindings

gahoekstraibm committed:
Merge branch 'master' of https://github.com/gahoekstraibm/WDR

gahoekstraibm committed:
Revert "Create bla.py"

This reverts commit 77422110b504e82c125938fe7e052af8196f1446.

gahoekstraibm committed:
Revert "Update util.py"

This reverts commit e1239a27353649d2ec84dcbb423d595eac7ed235.

Revert "Update util.py"

This reverts commit e4dcd075ebfe18fe02f33651fa88227f7ca73b08.

Revert "Update util.py"

This reverts commit fdd1acffdc04531f7fe4a564b21bafd28b5a9de3.

Revert "Update util.py"

This reverts commit 74a256f1fe3ec2dac7dd13db74a6129fe1ee0cbf.

Revert "Update util.py"

This reverts commit f2b07bad3079065d839dffaf7c0313df5166d17c.

Revert "Update util.py"

This reverts commit 045b1b2240363f6459e839d9d6d50b6b6b72c0b7.

Revert "Update util.py"

This reverts commit d7cb932d6fa9fee234914360a9aa1dbed5eeb283.

Revert "Update util.py"

This reverts commit f182f8b396d298110bde99d0907c88843105b196.

Revert "implemented new checksum generation for EARs"

This reverts commit 8fc93fffff0c183a02e71961841ed7d53f23a59f.

gahoekstraibm committed:
Corrected merge issue for internal variables

gahoekstraibm committed:
Update manifest.py

Corrected merge issue for internal variables

gahoekstraibm committed:
Corrected merge issue for internal variables

gahoekstraibm committed:
Corrected indentation

mplonka reviewed (commented):
Thanks for the pull-request!
Do you have any idea how to avoid ValueErrors while importing application manifests exported using current (pre-PR) version of WDR?

gahoekstraibm commented:
Hi Marcin, thanks for considering this pull request.
I will process the comments regarding white-space and python lists.
Also the remark regarding backwards compatibility is a valid one, I will look into this. Do you happen to have an example application manifest containing WS-policy bindings without policy type and attributes?

gahoekstraibm committed:
Updated after review comments from Marcin:
- removed some trailing white-space
- using Jython lists consistently
- Make WS Policy bindings backwards compatible

gahoekstraibm commented:
Hi Marcin, I think I made it downwards compatible now.
